code/game/area/areas.dm

```diff
@@ -278,11 +278,24 @@ GLOBAL_LIST_EMPTY(teleportlocs)
 /area/Destroy()
 	if(GLOB.areas_by_type[type] == src)
 		GLOB.areas_by_type[type] = null
-	GLOB.sortedAreas -= src
-	GLOB.areas -= src
+	//this is not initialized until get_sorted_areas() is called so we have to do a null check
+	if(!isnull(GLOB.sortedAreas))
+		GLOB.sortedAreas -= src
+	//just for sanity sake cause why not
+	if(!isnull(GLOB.areas))
+		GLOB.areas -= src
+	//turf cleanup
+	contained_turfs.Cut()
+	turfs_to_uncontain.Cut()
+	contained_turfs = null
+	turfs_to_uncontain = null
```

Comment:
Why Cut() and then set it to null? Why not just null it out in the first place?

Reply:
I guess i'm over cautious about memory leaks. I'll do that